registry/remote/credentials/internal/config/config.go

```diff
@@ -128,7 +129,23 @@ func Load(configPath string) (*Config, error) {
 
 	// decode config content if the config file exists
 	if err := json.NewDecoder(configFile).Decode(&cfg.content); err != nil {
-		return nil, fmt.Errorf("failed to decode config file at %s: %w: %v", configPath, ErrInvalidConfigFormat, err)
+		if !errors.Is(err, io.EOF) {
+			return nil, fmt.Errorf("failed to decode config file %s: %w: %v", configPath, ErrInvalidConfigFormat, err)
+		}
+
+		// Check to see if the credential file was an empty file
+		fileInfo, statErr := configFile.Stat()
+		if statErr != nil || fileInfo.Size() != 0 {
+			fileContent, readErr := os.ReadFile(configPath)
+			if readErr != nil || strings.TrimSpace(string(fileContent)) != "" {
+				return nil, fmt.Errorf("failed to decode config file %s: %w: %v", configPath, ErrInvalidConfigFormat, err)
```

Comment:
The only case that `Decode` returns io.EOF is all whitespace, so this code looks good, but is not testable. In the case of a file containing `{` for example, it will return "Unexpected EOF" as reproduced in the test below. Keep this code or get rid of it?

Reply:
> In the case of a file containing { for example, it will return "Unexpected EOF" as reproduced in the test below.

A file containing `{` should be considered corrupted.
The implementation in #962 is actually cleaner and can pass all tests.

Reply:
There is no need to stat the file because empty or just whitespace are the only cases that return `io.EOF`. If there are any other characters in the file it returns `io.ErrUnexpectedEOF`